src/Avalonia.Controls/Grid.cs

```diff
@@ -486,456 +279,138 @@ protected override Size MeasureOverride(Size constraint)
         /// <returns>The space taken.</returns>
         protected override Size ArrangeOverride(Size finalSize)
         {
-            int colCount = ColumnDefinitions.Count;
-            int rowCount = RowDefinitions.Count;
-            int colMatrixDim = _colMatrix.GetLength(0);
-            int rowMatrixDim = _rowMatrix.GetLength(0);
-
-            RestoreMeasureResults();
+            // Situation 1/2:
+            // If the grid doesn't have any column/row definitions,
```

Comment:
"If the grid doesn't have any column/row definitions, it behaves like a normal panel."